sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -17,13 +17,17 @@
 
 package org.apache.spark.sql
 
-import java.io.CharArrayWriter
+import java.io.{ByteArrayOutputStream, CharArrayWriter}
+import java.nio.channels.Channels
 
 import scala.collection.JavaConverters._
 import scala.language.implicitConversions
 import scala.reflect.runtime.universe.TypeTag
 import scala.util.control.NonFatal
 
+import org.apache.arrow.memory.RootAllocator
+import org.apache.arrow.vector.file.ArrowWriter
+import org.apache.arrow.vector.schema.ArrowRecordBatch
```

Comment:
It may be more palatable to encapsulate these details in a class that is internal to Spark